arch/xtensa/core/syscall_helper.c

```diff
@@ -27,3 +28,22 @@ uintptr_t xtensa_syscall_helper(uintptr_t arg1, uintptr_t arg2,
 
 	return a2;
 }
+#endif /* CONFIG_XTENSA_SYSCALL_USE_HELPER */
+
+#if XCHAL_HAVE_THREADPTR == 0
+#include <xtensa/config/core-isa.h>
+#include <xtensa/config/core.h>
+
+bool xtensa_is_user_context(void)
```

Comment:
I think this is used frequent enough to warrant special treatment. If this is implemented as a normal syscall, it will need to go through all the level 1 exception code (e.g. cross stack, calling in C function, marshaling of args, etc.).

TBH, here it is to cover our basis just in case such configuration exists...